cranelift/codegen/src/isa/aarch64/lower.rs

```diff
@@ -438,7 +446,20 @@ pub(crate) fn put_input_in_rse_imm12<C: LowerCtx<I = Inst>>(
 ) -> ResultRSEImm12 {
     if let Some(imm_value) = input_to_const(ctx, input) {
         if let Some(i) = Imm12::maybe_from_u64(imm_value) {
-            return ResultRSEImm12::Imm12(i);
+            let out_ty_bits = ty_bits(ctx.output_ty(input.insn, 0));
+            let ty_signed_max = if out_ty_bits <= 1 {
+                out_ty_bits as i64
+            } else {
+                (1i64 << (out_ty_bits - 1)) - 1
+            };
+            let is_negative = i.bits > (ty_signed_max as u16);
+
+            // This condition can happen if we matched a value that overflows the output type of
+            // its `iconst` when viewed as a signed value (i.e. iconst.i8 200).
```

Comment:
This feels somehow like the wrong place to be handling this. IMHO, `iconst.i8 200` should be rejected by the CLIF validator, because the given constant value is out-of-range for the `i8` type. Instead we're saying here that we should implement a wrapping behavior, but then we're only doing it in one backend (so then really we should address it in all backends instead).

I'm curious what others think, but to me this seems like a weird corner of IR semantics that is safer to define as "disallowed"...

Reply:
(Nevermind this, resolving comment.)